### ISSUE TITLE
Fluent API support creating Load Balancer with just NAT port forwarding rules (no load balance rule)

### ISSUE THREAD

**Issue by jefflill** [Azure/azure-sdk-for-net#3173]
I'd like to create a load balancer with only NAT port forwarding rules but without any load balancing rules.  This is not currently supported by the fluent API.

Here's what I'd like to be able to do:

````
using System;

using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
using Microsoft.Azure.Management.Network.Fluent.Models;

namespace AzureMultpleRoutes
{
    class Program
    {
        static void Main(string[] args)
        {
            // NOTE: You'll need to replace these with your own Azure credentials.

            const string clientId       = "****";
            const string clientSecret   = "****";
            const string tenantId       = "****";
            const string subscriptionId = "****";

            const string resourceGroup  = "azure-bug";
            const string region         = "westus2";

            // Login the API.

            var azureCredentials =
                new AzureCredentials(
                    new ServicePrincipalLoginInformation()
                    {
                        ClientId     = clientId,
                        ClientSecret = clientSecret
                    },
                tenantId,
                AzureEnvironment.AzureGlobalCloud);

            var azure = Azure.Configure()
                .Authenticate(azureCredentials)
                .WithSubscription(subscriptionId);

            // Create the resource group if necessary.

            if (!azure.ResourceGroups.CheckExistence(resourceGroup))
            {
                azure.ResourceGroups
                    .Define(resourceGroup)
                    .WithRegion(region)
                    .Create();
            }

            // Create a load balancer with inbound NAT rules but no load balancing rule.

            var pipLb = azure.PublicIPAddresses
                .Define("load-balancer-pip")
                .WithRegion(region)
                .WithExistingResourceGroup(resourceGroup)
                .WithDynamicIP()
                .WithLeafDomainLabel("lb-" + Guid.NewGuid().ToString())
                .Create();

            azure.LoadBalancers
                .Define("load-balancer")
                .WithRegion(region)
                .WithExistingResourceGroup(resourceGroup)
                .DefinePublicFrontend("frontend")
                   .WithExistingPublicIPAddress(pipLb)
                   .Attach()
                .DefineInboundNatRule("nat-0")          // <----- Compiler Error: "IWithPublicFrontendOrBackend does not contain a definition for 'DefineInboundNatRule'..."
                    .WithProtocol(TransportProtocol.Tcp)
                    .WithFrontend("frontend")
                    .WithFrontendPort(1000)
                    .WithBackendPort(2000)
                    .WithIdleTimeoutInMinutes(5)
                    .Attach()
                .DefineInboundNatRule("nat-1")
                    .WithProtocol(TransportProtocol.Tcp)
                    .WithFrontend("frontend")
                    .WithFrontendPort(1001)
                    .WithBackendPort(2002)
                    .WithIdleTimeoutInMinutes(5)
                    .Attach()
                .Create();
        }
    }
}
````
I get a compiler error when I call `DefineInboundNatRule()` without first calling `DefineLoadBalancingRule()` in the fluent call chain.

**Comment by martinsawicki** [comment]
yes, it's on the to-do list to rework the LB model to allow for such scenarios among others.

**Comment by hovsepm** [comment]
@jefflill Available in v1.1.2. Please update your package reference.
